jvb/src/main/kotlin/org/jitsi/videobridge/octo/OctoRelayService.kt

```diff
@@ -22,84 +22,79 @@ import org.jitsi.videobridge.transport.octo.BridgeOctoTransport
 import org.jitsi.videobridge.transport.udp.UdpTransport
 import org.jitsi.videobridge.util.TaskPools
 import java.net.SocketAddress
-import java.net.SocketException
-import java.net.UnknownHostException
 import java.time.Instant
 
+/**
+ * The service which is responsible for sending and receiving packets on the Octo link.  Only
+ * a single instance exists, and it handles all Octo traffic.  It should _only_ be created
+ * when it has been enabled in configuration.  It will throw in the constructor if it
+ * is not enabled
+ */
 class OctoRelayService {
     /**
      * The [UdpTransport] used to send and receive Octo data
      */
-    private var udpTransport: UdpTransport? = null
+    private val udpTransport: UdpTransport
 
     /**
      * The [BridgeOctoTransport] for handling incoming and outgoing Octo data
      */
-    var bridgeOctoTransport: BridgeOctoTransport? = null
-        private set
+    val bridgeOctoTransport: BridgeOctoTransport
 
-    fun start() {
+    init {
         if (!config.enabled) {
-            logger.info("Octo relay is disabled")
-            return
+            throw IllegalStateException("Octo relay service is not enabled")
         }
 
         val address = config.bindAddress
         val publicAddress = config.publicAddress
         val port = config.bindPort
 
-        try {
-            udpTransport = UdpTransport(address, port, logger, OCTO_SO_RCVBUF, OCTO_SO_SNDBUF)
-            logger.info("Created Octo UDP transport")
-        } catch (e: Exception) {
-            when (e) {
-                is UnknownHostException, is SocketException -> {
-                    logger.error("Failed to initialize Octo UDP transport with " +
-                            "address " + address + ":" + port + ".", e)
-                    return
-                }
-                else -> throw e
-            }
-        }
+        udpTransport = UdpTransport(address, port, logger, OCTO_SO_RCVBUF, OCTO_SO_SNDBUF)
+        logger.info("Created Octo UDP transport")
```

Comment:
Can this no longer throw?  Or is something outside this constructor catching it?

Reply:
It can, but I wanted it to bubble up.  I forgot we still create this in the singleton, though, so I think we will still need some form of this + a try catch in the singleton